source/default/_patterns/02-molecules/utccard/_utccard.twig

```diff
@@ -0,0 +1,95 @@
+ *   - id: A valid HTML ID and guaranteed unique.
+ * - title_attributes: Same as attributes, except applied to the main title
+ *   tag that appears in the template.
+ * - title_prefix: Additional output populated by modules, intended to be
+ *   displayed in front of the main title tag that appears in the template.
+ * - title_suffix: Additional output populated by modules, intended to be
+ *   displayed after the main title tag that appears in the template.
+ *
+ * @see template_preprocess_block()
+ * 
+ * This is the PL version of the block--utc-card-base.html.html.twig. NOTE: The image is currently hardcoded.
+ */
+#}
+
+{% set card_classes = [
+  'utc-card-2',
+  'h-100',
+  utccard.card_type ? 'utc-card-2--' ~ utccard.card_type,
+  utccard.card_color ? 'utc-card-2--' ~ utccard.card_color : 'utc-card-2--white',
+  utccard.card_width ? 'utc-card-2--w-' ~ utccard.card_width,
+  utccard.text_alignment ? 'utc-card-2--align-' ~ utccard.text_alignment,
+  utccard.card_link ? 'utc-card-2--card-link',
+  utccard.card_img_fill ? card_img_fill
+] | sort | join(' ') | trim %}
+
+{# could use include statements for items like images and buttons in design system #}
+  {% if utccard.card_link %}
+    <a href="{{ utccard.card_link }}" target="{{ utccard.card_link_target }}">
+  {% endif %}
+  
+
+  <div class="{{ card_classes }} max-w-md">
+
+    {% if (utccard.card_icon is empty) and utccard.card_image %}
+      <img src='http://placeimg.com/900/600/business'>
+    {% endif %}
+
+    {% if utccard.card_icon %}
+      <div class="utc-card-2__icon utc-card-2__icon--{{ utccard.card_icon_color }}">
+        {{ utccard.card_icon }}
+      </div>
+    {% endif %}
+    {% block card_body %}
+        <div class="utc-card-2__body">
+          <div class="utc-card-2__content">
+            {% if utccard.card_title %}
+              <h3 class="utc-card-2__title">{{ utccard.card_title }}
+              </h3>
+            {% endif %}
+            <p class="font-utcheadings">This card features the following:</p>
+            <ul class="pl-0">
+            {% if utccard.card_text %}
+            <div class="utc-card-2__text mb-8">
+                {% for item in utccard.card_text %}
+                  <li class="font-utcheadings py-0 my-0 block">
+                    {{ item }}
+                  </li>
+                {% endfor %}
+            </div>
+            {% endif %}
+             </ul>
+          </div>
+      
+        {% if (utccard.card_link is empty) and utccard.card_button_url %}
+          <div class="utc-card-2__action-container">
+            <a href="{{ utccard.card_button_url }}" class="font-medium btn {{ utccard.card_button_style }}" target="{{ utccard.card_button_target }}">
+              {{ utccard.card_button_text }}
+           </a>
+         </div>
+        {% endif %}
+
+       </div>
+      {% endblock card_body %}
+
+  </div>
+
+	{% if utccard.card_link %}
+     </a>
+  {% endif %}
```

Comment:
Are there future plans to take this and make it tailwind vs. the css classes?  Maybe in further redesigns? I do know I've found some things to be trickier in tailwind (e.g. the way this one seamlessly modifies classes for alignment, width, etc. based on user input without having to use a bunch of if statements)...and some things to be easier. Just curious. This one was an OG from pre-tailwind days.

Reply:
I think your suggestions are great! Thank you! I just wasn't sure if I needed to include all options bc if developers need the code from "Expand" button, they might have to recode it themselves if they want alternative solutions.

I used tailwinds where I could, but left existing classes to keep the integrity of the existing layout. I can look at converting them, though. Thoughts @bmartinez287?

Reply:
I'll take a look at the button taxonomy and see what I can do there.

Reply:
So, it be great if could simplify and refactor some of the code. I think a feature as complex and as useful as cards could use it. It be great exploring which items could be updated to tailwindcss and which ones might be too hard to rewrite/update.

In regards to the pattern-lab demo, we can probably break a card into multiple card types and then put them together based on certain if statements. Like sideimagecard.twig, topimagecard.twig
That should make cards easier to visualise and maybe reuse in code. We probably need to do more reasearch into standard naming conventions and way others are handling such complex items.

Reply:
I'm also wondering... if we do the buttons in the atoms, maybe we don't need all the button examples because developers would know (and we could add instructions) to replace any button code with their new button code. That would eliminate a good bit of the cards and keep the page focused on actual card variance.

Reply:
Yeah, i don't we need the text varient or buttons or other smaller items. We just need the large layout variations.

Reply:
@UTCbridget let me know if you need anything from me if you decide to refactor the cards with some more tailwind. Happy to help, but also happy to let you take the lead on it if it's a useful way for you to get to know that element and what could be done to improve it! Just lmk either way.

Reply:
@brittrich, could you do the buttons? I'm going to redo the cards without all the button variations. I'll look at the refractoring and see what can be done there.

Reply:
Sounds good to me! I'll work on buttons next.

Reply:
I have reworked the PL Cards with different variations combined per our conversation above and separated them out into Horizontal and Vertical cards. Instructions have been provided, indicating that card features can be mixed and matched. I have also refactored the cards to TailwindCSS. 

**Vertical Cards:** 

![utccard-pl-component-vt](https://user-images.githubusercontent.com/82905787/129774653-554be88f-e959-4399-a7ca-675b0ff7bf45.gif)

**Horizontal Cards:**

![utccard-pl-component-hz](https://user-images.githubusercontent.com/82905787/129774707-e997416c-cc41-4522-b6ee-a6d5638a862f.gif)

Branded Welcome: 

![Screen Shot 2021-08-17 at 1 10 32 PM](https://user-images.githubusercontent.com/82905787/129774741-c2e2f068-3682-4db8-b4d8-9917674db6ff.png)

I look forward to your thoughts!

Reply:
I think this looks great, Bridget! 

Last thought...and it may not be a useful one....but what if the Titles of the Cards reflected the key feature that's being demoed? For example, Left Align / Center Align / Right Align...or....75% Width Card / 50% Width Card / etc.

I've been looking at this design system some for the way they do the buttons. The way they handle cards is also interesting, looks like they have alot of variations as well: https://boltdesignsystem.com/pattern-lab/?p=viewall-components-card 

Can I ask, PL newb that I am, what's the purpose of the code snip/example in the context of our design system? Is it so that folks can use/recreate the component in another system, perhaps? Or, so we can see how everything connects in the code at a glance?

I'm learning a lot from this PR and Bernardo's PR for the Hero as I work on getting the buttons added! Hopefully will have something to show there by the end of this week that can maybe be integrated up to the Cards.

Reply:
It has a few.

1. Provide consistent design for Drupal and WordPress as they would tie into the same components.
2. Provide a kitchen sink of features for vendors such as https://gomocs.com/  https://securelb.imodules.com/s/1341/utc/form/19/form.aspx?sid=1341&gid=3&pgid=4310&cid=6419
3. Allow IT and other tech-based units on campus to have a visual representation of branding and a living example they could try to use or imitate. 


What Figma does for graphic designers on campus, pattern lab will do for developers, vendors, and internal resources.
Figma is supposed to give the flexibility to make quick visual changes based on a designer's input.

Patternlab might even be useful for some people on campus as they might be able to see some sample page layouts for homepages and choose if they want to recreate those or do their own.

Reply:
Nice! Makes sense! Thanks, Bernardo.